src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -741,6 +741,15 @@ - (MTRBaseDevice *)newBaseDevice
 
 @end
 
+@implementation MTRDevice (Deprecated)
+
++ (instancetype)deviceWithNodeID:(uint64_t)nodeID deviceController:(MTRDeviceController *)deviceController
```

Comment:
Same comment about `instancetype`, and you probably want to just call the non-deprecated method via `[self deviceWithNodeID:...]`

Reply:
Did the self thing.  Tracking the instancetype bit in https://github.com/project-chip/connectedhomeip/issues/23384